src/values/instruction_value.rs

```diff
@@ -282,6 +298,15 @@ impl<'ctx> InstructionValue<'ctx> {
         Ok(())
     }
 
+    // SubTypes: Only apply to alloca instruction
+    /// Returns the type that is allocated by the alloca instruction.
+    pub fn get_allocated_type(self) -> Result<AnyTypeEnum<'ctx>, &'static str> {
```

Comment:
Wouldn't this only be a BasicTypeEnum? I would imagine AnyType has types that cannot be allocated?

Reply:
Oh yeah that's correct! I incorrectly assumed it would be the same return type as get_type.